### ISSUE TITLE
[BUG] RandomIntervalClassifier giving FutureWarnings

### ISSUE THREAD

**Issue by jkrick** [sktime/sktime#5881]
**Describe the bug**

I am getting many of these warnings when running RandomIntervalClassifier on a pandas multi index  df.
`/opt/conda/lib/python3.11/site-packages/sktime/transformations/panel/catch22.py:326: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  series = np.array(X[i])`

**To Reproduce**

```python
#setup the classifier
clf = RandomIntervalClassifier(n_intervals = 20, n_jobs = -1, random_state = 43)

#fit the classifier on the training dataset
clf.fit(X_train, y_train)

```

**Expected behavior**

**Additional context**

Here is the output of 
'check_is_mtype(X_train, mtype="pd-multiindex", scitype="Panel", return_metadata=True)'

(True,
 None,
 {'is_univariate': False,
  'is_empty': False,
  'has_nans': False,
  'n_features': 10,
  'feature_names': ['time',
   'flux_panstarrs_g',
   'flux_panstarrs_i',
   'flux_panstarrs_r',
   'flux_panstarrs_y',
   'flux_panstarrs_z',
   'flux_w1',
   'flux_w2',
   'flux_zg',
   'flux_zr'],
  'n_instances': 264,
  'is_one_series': False,
  'is_equal_length': True,
  'is_equally_spaced': True,
  'n_panels': 1,
  'is_one_panel': True,
  'mtype': 'pd-multiindex',
  'scitype': 'Panel'})

**Versions**
<details>

System:
    python: 3.11.6 | packaged by conda-forge | (main, Oct  3 2023, 10:40:35) [GCC 12.3.0]
executable: /opt/conda/bin/python
   machine: Linux-4.19.0-13-amd64-x86_64-with-glibc2.35

Python dependencies:
          pip: 23.3.1
       sktime: 0.25.0
      sklearn: 1.3.2
       skbase: 0.6.2
        numpy: 1.26.0
        scipy: 1.11.3
       pandas: 2.1.3
   matplotlib: 3.8.1
       joblib: 1.3.2
        numba: 0.58.1
  statsmodels: 0.14.0
     pmdarima: None
statsforecast: None
      tsfresh: None
      tslearn: None
        torch: None
   tensorflow: None
tensorflow_probability: None

</details>
